widget-function/widgetfn/handler.py

```diff
@@ -23,7 +23,10 @@ def get_widget(event, context={}):
     encoded_image = base64.b64encode(widget_img["MetricWidgetImage"])
 
     res["statusCode"] = 200
-    if event["queryStringParameters"]["asstring"] == "true":
+    if (
+        "asstring" in event["queryStringParameters"]
+        and event["queryStringParameters"]["asstring"] == "true"
```

Comment:
:+1: on the update

do we know if this will strictly be a runtime `bool` or does it need to handle true-like values as well? If the former, I believe is `True/False/None` is generally considered to be more idiomatic

```python
if x is True
```

Reply:
it is a string because it comes in as a query param from the url, in that context in the lambda everything on `queryStringParameters` is a string, the same params are represented as arrays in `multiQueryStringParameters` but those are the only 2 types

Reply:
in a larger project I would for sure have a utility function to convert these to real types

Reply:
gotcha. wasn't sure if there was any type mapping, as many popular http client and server libs to the type mapping for you